### ISSUE TITLE
Allow custom Formatters for sbt console

### ISSUE THREAD

**Issue by matanster** [lihaoyi/utest#77]
It would be very very nice allowing the user to have their own output format in sbt, given that test results are plain scala data exposed to user code.

Currently one gets  something like 

```
[info] 1/4     utest.MyTestSuite.hello.world        Success((1,2))
[info] 2/4     utest.MyTestSuite.hello      Success
[info] 3/4     utest.MyTestSuite.test2      Failure(utest.AssertionError: assert(a == b)
[info] ...
[info] 4/4     utest.MyTestSuite.       Success
[info] -----------------------------------Results-----------------------------------
[info] MyTestSuite$     Success
[info]     hello        Success
[info]         world        Success((1,2))
[info]     test2        Failure(utest.AssertionError: assert(a == b)
[info]     ...
[info] Tests: 4
[info] Passed: 3
[info] Failed: 1
```

Once `sbt test` finishes off; which is, mostly a nuisance if one wishes to have their own output format in `sbt`, or just improve this default. Unfortunately I hadn't managed working on the code in eclipse as it doesn't import in Scala-IDE, otherwise I'd work on a patch.

**Comment by matanster** [comment]
I think this will also enable people who are used to the output format of legacy test frameworks, easily reproduce those formats with utest.

**Comment by matanster** [comment]
I could alternatively address the intent differently, by asking how would you typically run the tests "programmatically" without sbt (or without the `sbt test` command set), without making the test code accessible to those who use your library.

**Comment by lihaoyi** [comment]
Fixed in the WIP `test` branch targeting 0.4.0

**Comment by matanster** [comment]
This would be awesome :+1: :)
